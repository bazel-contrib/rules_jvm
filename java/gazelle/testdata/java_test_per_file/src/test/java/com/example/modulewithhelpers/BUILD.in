# gazelle:java_module_granularity module
