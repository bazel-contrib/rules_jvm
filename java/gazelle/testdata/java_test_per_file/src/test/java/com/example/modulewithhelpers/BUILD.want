load("@rules_java//java:defs.bzl", "java_library", "java_test")

java_library(
    name = "modulewithhelpers",
    srcs = [
        "Helper.java",
        "withdirectory/Helper.java",
    ],
    _gazelle_imports = [
        "com.example.myproject.App", # TODO: per-file tests should have per-test imports, excluding this one
        "com.google.common.math.IntMath",
        "java.lang.Exception", # TODO: per-file tests should have per-test imports, excluding this one
        "org.hamcrest.MatcherAssert", # TODO: per-file tests should have per-test imports, excluding this one
        "org.hamcrest.Matchers", # TODO: per-file tests should have per-test imports, excluding this one
        "org.junit.Assert", # TODO: per-file tests should have per-test imports, excluding this one
        "org.junit.Test", # TODO: per-file tests should have per-test imports, excluding this one
    ],
    _java_packages = [
        "com.example.modulewithhelpers",
        "com.example.modulewithhelpers.withdirectory",
    ],
    visibility = ["//:__subpackages__"],
)

java_test(
    name = "com_example_modulewithhelpers_AppTest",
    srcs = ["AppTest.java"],
    _gazelle_imports = [
        "com.example.modulewithhelpers",
        "com.example.myproject.App",
        "com.google.common.math.IntMath", # TODO: per-file tests should have per-test imports, excluding this one
        "java.lang.Exception",
        "org.hamcrest.MatcherAssert", # TODO: per-file tests should have per-test imports, excluding this one
        "org.hamcrest.Matchers", # TODO: per-file tests should have per-test imports, excluding this one
        "org.junit.Assert",
        "org.junit.Test",
    ],
    _java_packages = ["com.example.modulewithhelpers"],
    test_class = "com.example.modulewithhelpers.AppTest",
)

java_test(
    name = "com_example_modulewithhelpers_withdirectory_AnotherTest",
    srcs = ["withdirectory/AnotherTest.java"],
    _gazelle_imports = [
        "com.example.modulewithhelpers.withdirectory",
        "com.example.myproject.App",
        "com.google.common.math.IntMath", # TODO: per-file tests should have per-test imports, excluding this one
        "java.lang.Exception",
        "org.hamcrest.MatcherAssert",
        "org.hamcrest.Matchers",
        "org.junit.Assert", # TODO: per-file tests should have per-test imports, excluding this one
        "org.junit.Test",
    ],
    _java_packages = ["com.example.modulewithhelpers.withdirectory"],
    test_class = "com.example.modulewithhelpers.withdirectory.AnotherTest",
)
