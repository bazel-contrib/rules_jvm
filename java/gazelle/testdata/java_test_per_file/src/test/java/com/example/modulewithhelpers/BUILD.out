load("@rules_java//java:defs.bzl", "java_library", "java_test")

# gazelle:java_module_granularity module

java_library(
    name = "modulewithhelpers",
    testonly = True,
    srcs = [
        "Helper.java",
        "withdirectory/Helper.java",
    ],
    deps = [
        "//src/main/java/com/example/myproject",
        "@maven//:com_google_guava_guava",
        "@maven//:junit_junit",
        "@maven//:org_hamcrest_hamcrest_all",
    ],
)

java_test(
    name = "com_example_modulewithhelpers_AppTest",
    srcs = ["AppTest.java"],
    test_class = "com.example.modulewithhelpers.AppTest",
    deps = [
        ":modulewithhelpers",
        "//src/main/java/com/example/myproject",
        "@maven//:com_google_guava_guava",
        "@maven//:junit_junit",
        "@maven//:org_hamcrest_hamcrest_all",
    ],
)

java_test(
    name = "com_example_modulewithhelpers_withdirectory_AnotherTest",
    srcs = ["withdirectory/AnotherTest.java"],
    test_class = "com.example.modulewithhelpers.withdirectory.AnotherTest",
    deps = [
        ":modulewithhelpers",
        "//src/main/java/com/example/myproject",
        "@maven//:com_google_guava_guava",
        "@maven//:junit_junit",
        "@maven//:org_hamcrest_hamcrest_all",
    ],
)
